test/config-flags-gulpfile.js

```diff
@@ -51,40 +52,50 @@ describe('config: flags.gulpfile', function() {
     }
   });
 
-  it('Should ignore a ./gulp.* file if another directory is specified by ' +
-  '\n\t--cwd', function(done) {
+  it('Should find up project dir and use config file there', function(done) {
+    var topProj = path.resolve(fixturesDir, './flags/gulpfile/top-prj');
+    var initCwd = path.resolve(topProj, 'sub-dir');
+
     runner
-      .chdir('./flags/gulpfile')
-      .gulp('--cwd ./cwd')
+      .chdir(initCwd)
+      .gulp('-T')
       .run(cb);
 
     function cb(err, stdout, stderr) {
       expect(err).toEqual(null);
       expect(stderr).toEqual('');
-      stdout = headLines(stdout, 1, 3);
-      expect(stdout).toEqual(
-        'Another gulpfile : ' +
-          path.join(fixturesDir, 'flags/gulpfile/cwd/gulpfile.js')
-      );
+      stdout = eraseTime(stdout, 1, 3);
+      expect(stdout.split(/\r\n|\r|\n/)).toEqual([
+        'Working directory changed to ' + tildify(topProj, '~'),
+        'Config with ' + path.resolve(topProj, '.gulp.js'),
+        '└── default  ' + path.resolve(topProj, 'gulpfile.js'),
+        '',
+      ]);
       done(err);
     }
   });
 
-  it('Should ignore a ./.gulp.* file if another gulpfile is specified by ' +
-  '\n\t--gulpfile', function(done) {
+  // This test case fails because of a bug that `flags.gullpfile` does not
+  // change process.cwd().
+  it.skip('Should use config file here and use gulpfile specified in config file', function(done) {
```

Comment:
@sttk I just went through this and I think it's too big of issue to ship these changes without having this working correctly.

I think it's a fundamental flaw in how we added config files to Liftoff, because they don't affect it's configBase, etc lookups when we rename the gulpfile.

I'm not sure how to solve this.